intellij/src/saros/intellij/ui/tree/ContactPopMenu.java

```diff
@@ -42,49 +44,108 @@
-
     for (Project project : ProjectManager.getInstance().getOpenProjects()) {
-      Pair<JMenu, List<String>> results = prepareProjectMenu(project);
+      menuShareProject.add(createProjectMenu(project));
+    }
 
-      if (results == null) {
-        return;
-      }
+    add(menuShareProject);
+  }
+
+  /**
+   * Prepares the JMenu for the given project. The menu contains entries for all modules that belong
+   * to the project. The entries are grouped by whether they are shareable or not. Non-shareable
+   * entries are disabled and have a tooltip explaining why/that they are not sharable.
+   *
+   * <p>If the module information for the given project can not be read, a disabled menu entry is
+   * returned instead. The entry has a tooltip explaining that there was an issue reading the module
+   * information.
+   *
+   * @param project the project whose menu to prepare
+   * @return the <code>JMenu</code> for the project or a disabled <code>JMenuEntry</code> containing
+   *     only the project name if the module information could not be obtained for the project
+   */
+  @NotNull
+  private JMenuItem createProjectMenu(@NotNull Project project) {
+
+    Pair<List<JMenuItem>, List<JMenuItem>> moduleItems = createModuleEntries(project);
+
+    if (moduleItems == null) {
+      JMenuItem errorMessageItem = new JMenuItem(project.getName());
+      errorMessageItem.setToolTipText(Messages.ContactPopMenu_menu_entry_error_processing_project);
+      errorMessageItem.setEnabled(false);
 
-      JMenu projectMenu = results.first;
-      List<String> projectNonCompliantModules = results.second;
+      return errorMessageItem;
+    }
+
+    JMenu projectMenu = new JMenu(project.getName());
 
-      menuShareProject.add(projectMenu);
+    List<JMenuItem> shownModules = moduleItems.first;
+    List<JMenuItem> nonCompliantModules = moduleItems.second;
 
-      if (!projectNonCompliantModules.isEmpty()) {
-        nonCompliantModules.addAll(projectNonCompliantModules);
+    if (!shownModules.isEmpty()) {
+      shownModules.sort(Comparator.comparing(JMenuItem::getText));
+
+      for (JMenuItem moduleItem : shownModules) {
+        projectMenu.add(moduleItem);
       }
+
+    } else {
+      LOG.debug(
+          "No modules shown to user as no modules "
+              + (nonCompliantModules.isEmpty()
+                  ? ""
+                  : "complying with our current release restrictions ")
+              + "were found");
+
+      JMenuItem noModulesFoundMenuItem =
+          new JMenuItem(
+              nonCompliantModules.isEmpty()
+                  ? Messages.ContactPopMenu_menu_entry_no_modules_found
+                  : Messages.ContactPopMenu_menu_entry_no_valid_modules_found);
+
+      noModulesFoundMenuItem.setEnabled(false);
+
+      projectMenu.add(noModulesFoundMenuItem);
     }
 
+    // Show project module as non-sharable
+    projectMenu.addSeparator();
+
+    JMenuItem projectItem = new JMenuItem(project.getName());
```

Comment:
I really do not know if this is an accepted way of building menu entries, i.e creating disabled menu entries describing the entries below such an entry.

Reply:
I think you misunderstood. Yes, the entry is disabled, but it is not used to describe other entries below it. Since in IntelliJ, the base of a project (containing the project config) is also just a module, I also wanted to display this module in the list. This part of the code adds an entry for the project module that is disabled and contains a tooltip explaining that the project module currently can not be shared.

Reply:
Ah ok my bad. But as this is always disabled I guess it would not hurt to do not display this entry at all. Of course that would / will require that the developers read the manual :D